django_mongodb/fields/json.py

```diff
@@ -0,0 +1,171 @@
+from django.db import NotSupportedError
+from django.db.models.fields.json import (
+    ContainedBy,
+    DataContains,
+    HasAnyKeys,
+    HasKey,
+    HasKeyLookup,
+    HasKeys,
+    JSONExact,
+    KeyTransform,
+    KeyTransformIn,
+    KeyTransformIsNull,
+    KeyTransformNumericLookupMixin,
+)
+
+from ..lookups import builtin_lookup
+from ..query_utils import process_lhs, process_rhs
+
+
+def contained_by(self, compiler, connection):  # noqa: ARG001
+    raise NotSupportedError("contained_by lookup is not supported on this database backend.")
+
+
+def data_contains(self, compiler, connection):  # noqa: ARG001
+    raise NotSupportedError("contains lookup is not supported on this database backend.")
+
+
+def _has_key_predicate(path, root_column, negated=False):
+    """Return MQL to check for the existence of `path`."""
+    result = {
+        "$and": [
+            # The path must exist (i.e. not be "missing").
+            {"$ne": [{"$type": path}, "missing"]},
```

Comment:
In what case would the path return missing? Wouldn't it always return null?

Reply:
> If the argument is a field that is missing in the input document, [$type](https://www.mongodb.com/docs/manual/reference/operator/aggregation/type/#mongodb-expression-exp.-type) returns the string "missing".